overlord/snapshotstate/backend/backend.go

```diff
@@ -330,7 +330,8 @@ func Save(ctx context.Context, id uint64, si *snap.Info, cfg map[string]interfac
 
 	w := zip.NewWriter(aw)
 	defer w.Close() // note this does not close the file descriptor (that's done by hand on the atomic writer, above)
-	if err := addDirToZip(ctx, snapshot, w, "root", archiveName, si.DataDir()); err != nil {
+	baseDir := snap.BaseDataDir(si.InstanceName())
+	if err := addSnapDirToZip(ctx, snapshot, w, "root", archiveName, baseDir, si.Revision); err != nil {
```

Comment:
We already have the revision in `snapshot.Revision`.

Reply:
Thank you, I've removed it now